solr/core/src/java/org/apache/solr/store/blob/client/LocalStorageClient.java

```diff
@@ -0,0 +1,259 @@
+      // with a real S3 implementation.
+      try (PrintWriter out = new PrintWriter(blobMetadataFile)){
+        out.println(json);
+      }  
+    } catch (Exception ex) {
+      throw new BlobException(ex);
+    }
+  }
+
+  @Override
+  public BlobCoreMetadata pullCoreMetadata(String sharedStoreName, String blobCoreMetadataName) throws BlobException {
+    try {
+      if (!coreMetadataExists(sharedStoreName, blobCoreMetadataName)) {
+        return null;
+      }
+      
+      String blobMetadataPath = getBlobAbsolutePath(getBlobMetadataName(sharedStoreName, blobCoreMetadataName));
+      File blobMetadataFile = new File(blobMetadataPath); 
+      
+      String json = new String(Files.readAllBytes(blobMetadataFile.toPath()));
+      ToFromJson<BlobCoreMetadata> converter = new ToFromJson<>();
+      return converter.fromJson(json, BlobCoreMetadata.class);
+    } catch (Exception ex) {
+      throw new BlobException(ex);
+    }
+  }
+
+  @Override
+  public boolean coreMetadataExists(String sharedStoreName, String blobCoreMetadataName) throws BlobException { 
+    try {
+      String blobMetadataPath = getBlobAbsolutePath(getBlobMetadataName(sharedStoreName, blobCoreMetadataName));
+      File coreMetadataFile = new File(blobMetadataPath); 
+      return coreMetadataFile.exists();
+    } catch (Exception ex) {
+      throw new BlobException(ex);
+    }
+  }
+
+  /**
+   * Prefixes the given path with the blob store root directory on the local FS 
+   */
+  private String getBlobAbsolutePath(String blobPath) {
+    return BlobClientUtils.concatenatePaths(blobStoreRootDir, blobPath);
+  }
+    
+  private void createCoreStorage(String blobName) throws Exception {
+    File coreRootDir = getCoreRootDir(blobName);
+    coreRootDir.mkdirs();
+    if (!coreRootDir.isDirectory()) {
+      throw new IOException("Can't create Blob core root directory " + coreRootDir.getAbsolutePath());
+    }
+  }
+    
+  @Override
+  public void deleteCore(String blobName) throws BlobException {
+    try {
+      Path path = Paths.get(getBlobAbsolutePath(blobName));
+      Files.walk(path)
+        // Since this traversal includes the root directory
+        // we need to reverse because we can't delete non-empty directories 
+        .sorted(Comparator.reverseOrder())
+        .map(Path::toFile)
+        .forEach(file -> {
+          file.delete();
+        });
+    } catch (Exception ex) {
+      // In case the path doesn't exist, we'll just swallow the exception because it's not an issue,
+      // especially in test clean up.
+    }
+  }
+    
+  /**
+   * On the local FS implementation we'll just delete blob files individually
+   */
+  @Override
+  public void deleteBlobs(Collection<String> paths) throws BlobException {
+    try {
+      for (String blobPath : paths) {
+        final File blobFile = new File(getBlobAbsolutePath(blobPath));
+        blobFile.delete();
```

Comment:
I believe it was in case of IOException. If the delete fails locally it's not really an issue, as file names are unique and the local file system is just to help dev/debug/tests, not meant to really be used.

Reply:
I see. But I can see people deploying NAS or NTFS mounts as shared index locations. As part of the community, I can see myself resorting to this. :) Other possibilities are HDFS and NoSQL like Cassandra, maybe? 

So, if local file system is not mean to be used in production there should be a clear way of stating/preventing this somehow. I feel people will probably (ab)use local file system as a blob storage once this PR is merged. /cc andyvuong